packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -335,7 +335,7 @@ const PickersDayRaw = React.forwardRef(function PickersDay<TDate>(
       centerRipple
       data-mui-test="day"
       disabled={disabled}
-      aria-label={!children ? utils.format(day, 'fullDate') : undefined}
+      aria-label={!children ? utils.format(day, 'fullDateWithWeekday') : undefined}
```

Comment:
```suggestion
```
This `aria-label` shouldn't even be needed once the roles are configured as suggested by ARIA example.